### PR TITLE
Start running tslint at the beginning of the process.

### PR DESCRIPTION
**Pull request by SebastianStehle**
I changed these things:

1. The plugin waits for the linter to complete.
2. The child process redirects the output to the parent as json and the parent parses the json and adds the result to the compilation object.

=> The tslint should produce the results faster and warning and error handling is respected.

Edit: I tested it in a bigger project of mine (https://github.com/squidex/squidex) and I realized that with the dev server the incremental build is much faster than the linting process and waiting for the results would slow down the development process. Therefore I added an option which allows to wait for the results and add the errors to the compilation errors/warnings or to keep it running in the background and write it to the output when it is ready (but not before the compilation has been completed).

### CODE REVIEW

File: src/linter.js [hunk 1]
```diff
@@ -1,50 +1,41 @@
 const { Runner, run } = require('tslint/lib/runner');
-const path = require('path');
-const chalk = require('chalk');
 
 const options = JSON.parse(process.argv[2]) || {};
 
-function log(message) {
-  if (!options.silent) {
```
Comment (jrparish) [inline src/linter.js:8]: We'll want to keep this option in case people want to suppress all output.

**Review by jrparish — COMMENTED** [review]
Thanks for doing this, I think it looks really good. 

- Can we keep the `silent` option? 
- Can we remove the `package-lock.json` files?

File: src/plugin.js [hunk 2]
```diff
@@ -4,38 +4,148 @@ const { fork } = require('child_process');
...
+      });
+    } else {
+      callback();
+    }
+  }
+
+  function doneHook() {
+    const currentIteration = linterIteration;
+    
+    if (linterPromise && !options.waitForLinting) {
```
Comment (jrparish) [inline src/plugin.js:116]: Can you document the new options?

File: src/linter.js (hunk 1 above)
Comment (SebastianStehle) [reply src/linter.js:8]: I know, I had it on my todo list to handle it in the plugin.js and have forgotten to add it. Will add it.

File: src/plugin.js (hunk 2 above)
Comment (SebastianStehle) [reply src/plugin.js:116]: Sure.

About the `package-lock.json`: I thought it is best practice to keep it in the source control.

**Comment by SebastianStehle** [comment]
I implemented your requests.

**Pull request merged**